pkg/resources/sos_bucket_policy/resource.go

```diff
@@ -0,0 +1,369 @@
+			"failed to create SOS client",
+			err.Error(),
+		)
+		return
+	}
+
+	if !plan.Policy.Equal(state.Policy) {
+		_, err = sosClient.PutBucketPolicy(ctx, &s3.PutBucketPolicyInput{
+			Bucket: plan.Bucket.ValueStringPointer(),
+			Policy: plan.Policy.ValueStringPointer(),
+		})
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"failed to put bucket policy",
+				err.Error(),
+			)
+			return
+		}
+	}
+
+	if err := pollBucket(ctx, sosClient, plan.Bucket.ValueString()); err != nil {
+		resp.Diagnostics.AddError(
+			"failed to poll bucket policy",
+			err.Error(),
+		)
+		return
+	}
+
+	state.Policy = plan.Policy
+
+	// Save updated state into Terraform state.
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+
+	tflog.Trace(ctx, "resource update done", map[string]interface{}{
+		AttrBucket: state.Bucket,
+	})
+}
+
+// Delete resources by receiving Terraform prior state data and performing deletion logic.
+func (r *ResourceSOSBucketPolicy) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state ResourceSOSBucketPolicyModel
+
+	// Load Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set timeout.
+	timeout, diags := state.Timeouts.Create(ctx, config.DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	sosClient, err := r.NewSOSClient(ctx, state.Zone.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"failed to create SOS client",
+			err.Error(),
+		)
+		return
+	}
+
+	_, err = sosClient.DeleteBucketPolicy(ctx, &s3.DeleteBucketPolicyInput{
+		Bucket: state.Bucket.ValueStringPointer(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"failed to put bucket policy",
+			err.Error(),
+		)
+		return
+	}
+
+	tflog.Trace(ctx, "resource deleted", map[string]interface{}{
+		AttrBucket: state.Bucket,
+	})
+}
+
+// ImportState lets Terraform begin managing existing infrastructure resources.
+func (r *ResourceSOSBucketPolicy) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, "@")
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		resp.Diagnostics.AddError(
+			"unexpected import identifier",
+			fmt.Sprintf("Expected import identifier with format: id@zone. Got: %q", req.ID),
+		)
+		return
+	}
+
+	var state ResourceSOSBucketPolicyModel
+
+	// Set timeouts (quirk https://github.com/hashicorp/terraform-plugin-framework-timeouts/issues/46)
+	var timeouts timeouts.Value
+	resp.Diagnostics.Append(resp.State.GetAttribute(ctx, path.Root("timeouts"), &timeouts)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	state.Timeouts = timeouts
+
+	state.Bucket = types.StringValue(idParts[0])
+	state.Zone = types.StringValue(idParts[1])
+	// TODO policy?
```

Comment:
This is fine, `Import` should only set attribute that are needed for `Read` to succeed.